nrich-notification/README.md

```diff
@@ -33,33 +47,78 @@ To be able to use this module following configuration is required:
-`String` for action notification.
+`NotificationResponseService` is a service that adds support for additional data in notifications by introducing additional methods that return `NotificationDataResponse`.
+When using Spring Web MVC users can use `NotificationResponseService` implementation `WebMvcNotificationResponseService` that
+is a wrapper around `NotificationResolverService` providing a convenient way of returning a notification with a response.
 
-`NotificationResponseService` is a service that adds support for returning data with notifications by introducing additional methods that return `NotificationDataResponse`.
+---
+
+### Default `MessageSource` setup
+
+Notification also includes `notificationMessages` property files that contain required default messages and should be added to
+`MessageSource` (**this is done automatically when using Spring Boot starter module**).
 
-When using Spring Web MVC users can use `NotificationResponseService` implementation `WebMvcNotificationResponseService` that is a wrapper around `NotificationResolverService` providing a
-convenient way of returning a notification with a response.
 
 ## Usage
 
-`nrich-notification` is used as a dependency of `nrich-webmvc` module. It is used in a `RestControllerAdvice` and on exceptions it creates notifications that are then sent to the client,
-so a common usage would be to use it in a `ControllerAdvice` to have unified error notification handling.
+Following examples of usage demonstrate `nrich-notification` module's ability to create notifications that respond to **actions**, **exceptions** and **validation errors**.
+`nrich-notification` is used as a dependency of [nrich-webmvc](../nrich-webmvc/README.md) module and as such is better explained with examples that are based on requests
+and responses.
+
+### Basic notification structure
+
+Every notification class in `nrich-notification` inherits `Notification` class that is specified in [nrich-notification-api](../nrich-notification-api/README.md) module and
+has these attributes:
+
+| Attribute                | Description                                       |
+|--------------------------|---------------------------------------------------|
+| `title`                  | Notification title.                               |
+| `content`                | Notification content.                             |
+| `messageList`            | List of additional messages.                      |
+| `severity`               | Indicates the importance of notification.         |
+| `uxNotificationOptions`  | Additional options that the client can interpret. |
+| `timestamp`              | Notification timestamp.                           |
+
+
+
+### Exception notifications
+
+If we wish to have unified error notification handling we can define `RestControllerAdvice` where, when handling exceptions with defined exception handlers,
```

Comment:
Here it is not clear that we already have such an advice maybe just mention that also?

Reply:
Mentioned!